packages/grid/src/GridRenderer.ts

```diff
@@ -3009,6 +3050,154 @@ export class GridRenderer {
 
     context.translate(-barLeft, -barTop);
   }
+
+  /**
+   * Gets the token boxes that are visible in the cell
+   * @param column The visible column
+   * @param row The visible row
+   * @param state The GridRenderState
+   * @returns An array of TokenBox of visible tokens or empty array
+   */
+  getTokenBoxesForVisibleCell(
+    column: VisibleIndex,
+    row: VisibleIndex,
+    state: GridRenderState
+  ): TokenBox[] {
+    const { metrics, context, model, theme } = state;
+    const { modelRows, modelColumns } = metrics;
+
+    if (context == null || metrics == null) {
+      return [];
+    }
+
+    const modelRow = getOrThrow(modelRows, row);
+    const modelColumn = getOrThrow(modelColumns, column);
+
+    const text = model.textForCell(modelColumn, modelRow);
+    const { width: textWidth, x: textX, y: textY } = this.getTextRenderMetrics(
+      state,
+      column,
+      row
+    );
+
+    const { fontWidths } = metrics;
+
+    // Set the font and baseline and change it back after
+    context.save();
+    context.font = theme.font;
+    context.textBaseline = 'middle';
+
+    const fontWidth =
+      fontWidths.get(context.font) ?? GridRenderer.DEFAULT_FONT_WIDTH;
+    const truncationChar = model.truncationCharForCell(modelColumn, modelRow);
+    const truncatedText = this.getCachedTruncatedString(
+      context,
+      text,
+      textWidth,
+      fontWidth,
+      truncationChar
+    );
+
+    const {
+      actualBoundingBoxAscent,
+      actualBoundingBoxDescent,
+    } = context.measureText(truncatedText);
+    const linkHeight = actualBoundingBoxAscent + actualBoundingBoxDescent;
+
+    const tokens = model.tokensForCell(
+      modelColumn,
+      modelRow,
+      truncatedText.length
+    );
+
+    context.restore();
+
+    // Check if the truncated text contains a link
+    if (tokens.length === 0) {
+      return [];
+    }
+
+    return this.getCachedTokenBoxesForVisibleCell(
+      truncatedText,
+      tokens,
+      theme.font,
+      'middle',
+      linkHeight,
+      context
+    ).map(tokenBox => ({
+      x1: tokenBox.x1 + textX,
+      y1: tokenBox.y1 + (textY - actualBoundingBoxAscent),
```

Comment:
I'm still seeing the token box is off by a couple pixels. We can debug together if you want, I'm not sure why it's off.